src/inmanta/server/services/compilerservice.py

```diff
@@ -388,6 +388,9 @@ async def request_recompile(
 
             :return: the compile id of the requested compile and any warnings produced during the request
         """
+        if env.halted:
+            return None, ["Skipping compile because environment has been halted."]
```

Comment:
This will ensure that no compile requests can be added to the queue, but all compile requests that were already present in the queue before the red button was pressed, will be executed. I don't think that's desired.

Reply:
Good catch

Reply:
We could go in several ways here:
* Clear the queue and ignore new compile requests
* Just ignore new compile requests and but keep the queue
* Keep queuing new compile requests, but don't execute them.

I am not sure what is best here.